### ISSUE TITLE
Scala3 found issues

### ISSUE THREAD

**Issue by kpbochenek** [scalameta/metals#1695]
In repo: https://github.com/kpbochenek/dotty-complex

---

Wrong hint?(mouse is on `Red`):
![2020-05-05-171336_541x117_scrot](https://user-images.githubusercontent.com/10478402/81083151-81426d00-8ef4-11ea-866e-e345e95e4814.png)

I think it should be something like: 
`case Red: com.kpbochenek.example.dotty.SimpleEnum.Red`

---

Wrong hint(mouse is on `Green`):
![2020-05-05-172315_740x99_scrot](https://user-images.githubusercontent.com/10478402/81083771-44c34100-8ef5-11ea-9295-a26a422f4109.png)

Funny enough hints are okay for other cases(except first).

---

Selecting `color` highlights whole (...) block and doesn't highlight all occurrences.
Also searching for references of `color` doesn't work.
![2020-05-05-173108_284x88_scrot](https://user-images.githubusercontent.com/10478402/81085348-68878680-8ef7-11ea-9b2f-1c8a07126b50.png)

Selecting a place where `color` is used correctly highlights everything(even declaration place)

---

Jumping to definition of String works most of time except this case, where it finds 3 definitions(because misinterprets it as `makeColor` function ??
![2020-05-05-180036_353x177_scrot](https://user-images.githubusercontent.com/10478402/81087721-59560800-8efa-11ea-8a5d-2a2fd23ab30d.png)

---

Highlight ends in half of next word :)
Also searching for references being on `Cancelable` yields no results, but searching on any other `Cancelable` yields 5 results
![2020-05-05-180802_351x66_scrot](https://user-images.githubusercontent.com/10478402/81088559-63c4d180-8efb-11ea-82be-84c0f698ed9c.png)

**Comment by tgodzik** [comment]
Thanks for checking it out, I think the first one I am already fixing in https://github.com/scalameta/metals/pull/1701

The rest is I think connected to how we generate semanticDB in Dotty and should be fixed there.

**Comment by kpbochenek** [comment]
just checked after #1701 merge and `IdentType` is fixed :+1: 
If rest of problems must be solved directly in dotty you can close this ticket

**Comment by tgodzik** [comment]
@kpbochenek  Maybe, let's keep this open or open a separate issue in Dotty? It's most likely still on our heads to fix it.

**Comment by tgodzik** [comment]
I will keep this open so that the errors are not lost. We might need to do the work in Dotty, but better to leave it here.
